package/network/utils/comgt/files/ncm.sh

```diff
@@ -117,21 +127,23 @@ proto_ncm_setup() {
 	}
 
 	echo "Starting network $interface"
-	json_get_vars connect
-	eval COMMAND="$connect" gcom -d "$device" -s /etc/gcom/runcommand.gcom || {
-		echo "Failed to connect"
-		proto_notify_error "$interface" CONNECT_FAILED
-		return 1
-	}
-
 	echo "Setting up $ifname"
-	
+
 	proto_init_update "$ifname" 1
 	proto_add_data
 	json_add_string "manufacturer" "$manufacturer"
 	proto_close_data
 	proto_send_update "$interface"
 
+	json_load "$(cat /etc/gcom/ncm.json)"
+	json_select "$manufacturer"
+	json_get_vars connect
+	eval COMMAND="$connect" gcom -d "$device" -s /etc/gcom/runcommand.gcom || {
+		echo "Failed to connect"
+		proto_notify_error "$interface" CONNECT_FAILED
+		return 1
+	}
+
```

Comment:
Hi matti,
I try the sequence manually after power on reset but it seems that the AT+CGDCONT shall be issued after the setpin.
```
AT+CFUN=1
OK
AT+CGDCONT=1,"IP","web.vodafone.de",0,0
ERROR
AT+CPIN?
+CPIN: SIM PIN

OK
AT+CPIN="****"
OK
AT+CGDCONT=1,"IP","web.vodafone.de",0,0
OK
AT+CGACT=1,1
OK
```
```
root@lede:~# uci set network.wanusb=interface
root@lede:~# uci set network.wanusb.ifname=usb0
root@lede:~# uci set network.wanusb.proto='dhcp'
root@lede:~# uci commit network.wanusb
root@lede:~# ifconfig
usb0      Link encap:Ethernet  HWaddr 02:00:FF:AA:AA:AA
          inet6 addr: fe80::ffff:feaa:aaaa/64 Scope:Link
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:1 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:0 (0.0 B)  TX bytes:110 (110.0 B)
```
```
AT+CGDATA="M-MBIM",1,1
OK
root@lede:~# ifconfig
usb0      Link encap:Ethernet  HWaddr 02:00:FF:AA:AA:AA
          inet addr:77.25.146.160  Bcast:77.25.146.191  Mask:255.255.255.192
          inet6 addr: fe80::ffff:feaa:aaaa/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:12 errors:0 dropped:0 overruns:0 frame:0
          TX packets:15 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:2886 (2.8 KiB)  TX bytes:1825 (1.7 KiB)
```

Shall I move the AT+CGDCONT in the connect or keep the configure state?